### PR TITLE
feat(gatekeepers): implement a gitcoin passport gatekeeper

### LINKED ISSUE #1444: Create a Gitcoin Passport gatekeeper
# Motivation

MACI polls have been run that e.g. enforce voters to hold a certain NFT or to have received a specific EAS attestation. We want to continue to expand the configurability of MACI operators to set eligibility requirements.

We think a great piece of functionality would be to have a Gitcoin Passport gatekeeper contract where operators can set certain criteria (e.g. a score threshold)

# Details

Create a `PassportGatekeeper` contract that can configure a Passport score threshold

### PR DESCRIPTION
**Pull request by ctrlc03**
# Description

Implement a gatekeeper that uses a gitcoin passport score as criteria to signup to MACI. 

## Related issue(s)

fix #1444 

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: contracts/contracts/gatekeepers/EASGatekeeper.sol [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { IEAS } from "../interfaces/IEAS.sol";
 /// only if they've received an attestation of a specific schema from a trusted attester
 contract EASGatekeeper is SignUpGatekeeper, Ownable(msg.sender) {
   // the reference to the EAS contract
-  IEAS private immutable eas;
+  IEAS public immutable eas;
```
Comment (kittybest) [inline contracts/contracts/gatekeepers/EASGatekeeper.sol:14]: why we make this private before, and public now?

File: contracts/contracts/interfaces/IGitcoinPassportDecoder.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL
+pragma solidity ^0.8.20;
+
+/// @dev A struct storing a passpor credential
```
Comment (kittybest) [inline contracts/contracts/interfaces/IGitcoinPassportDecoder.sol:4]: typo: passport

File: contracts/contracts/gatekeepers/GitcoinPassportGatekeeper.sol [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  /// @param _maci The MACI contract interface to be stored
+  function setMaciInstance(address _maci) public override onlyOwner {
+    if (_maci == address(0)) revert ZeroAddress();
+    maci = _maci;
+  }
+
+  /// @notice Register an user based on their attestation
+  /// @dev Throw if the attestation is not valid or just complete silently
+  /// @param _user The user's Ethereum address.
+  function register(address _user, bytes memory /*_data*/) public override {
```
Comment (kittybest) [inline contracts/contracts/gatekeepers/GitcoinPassportGatekeeper.sol:53]: what does the /*_data*/ do here?

**Review by kittybest — COMMENTED** [review]
thanks!

File: contracts/contracts/gatekeepers/EASGatekeeper.sol (hunk 1 above)
Comment (ctrlc03) [reply contracts/contracts/gatekeepers/EASGatekeeper.sol:14]: not sure why it was private, maybe to not allow to auto generate the public view function eas() - but I saw the hats gatekeeper had them public so thought why not keep them all public. If you remember once we deployed with this value wrong and had issues troubleshooting, now it should be easier 🤞

File: contracts/contracts/gatekeepers/GitcoinPassportGatekeeper.sol (hunk 3 above)
Comment (ctrlc03) [reply contracts/contracts/gatekeepers/GitcoinPassportGatekeeper.sol:53]: it's to not have a complain regarding the parameter not being used - but still follow the function signature which wants two parameters

File: contracts/tasks/deploy/maci/02-gatekeepers.ts [hunk 4]
```diff
@@ -72,5 +76,39 @@ deployment.deployTask("full:deploy-gatekeepers", "Deploy gatekeepers").then((tas
         network: hre.network.name,
       });
     }
+
+    const isSupportedGitcoinGatekeeperNetwork = ![
+      ESupportedChains.Hardhat,
+      ESupportedChains.Coverage,
+      ESupportedChains.Sepolia,
```
Comment (0xmad) [inline contracts/tasks/deploy/maci/02-gatekeepers.ts:83]: Why sepolia is not supported?

**Review by 0xmad — COMMENTED** [review]
@ctrlc03 thanks, just one question.

File: contracts/tasks/deploy/maci/02-gatekeepers.ts (hunk 4 above)
Comment (ctrlc03) [reply contracts/tasks/deploy/maci/02-gatekeepers.ts:83]: https://docs.passport.xyz/building-with-passport/smart-contracts/contract-reference#decoder-contract-addresses 
They only have it on op-sepolia and linea

**Review by 0xmad — APPROVED** [review]

**Pull request merged**